include/drjit-core/half.h

```diff
@@ -0,0 +1,117 @@
+/*
+    drjit-core/half.h -- minimal half precision number type
+
+    All rights reserved. Use of this source code is governed by a BSD-style
+    license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include <drjit-core/jit.h>
+#include <drjit-core/traits.h>
+#include <drjit-core/array.h>
+
+#include <cmath>
+#include <cstring>
+
+#if defined __aarch64__
+#include <arm_fp16.h>
+#elif defined(__F16C__)
+#include <emmintrin.h>
+#include <immintrin.h>
+#endif
+
+
+NAMESPACE_BEGIN(drjit)
+struct dr_half;
+NAMESPACE_END(drjit)
+
+NAMESPACE_BEGIN(std)
+template<> struct is_floating_point<drjit::dr_half> : true_type { };
+template<> struct is_arithmetic<drjit::dr_half> : true_type { };
+template<> struct is_signed<drjit::dr_half> : true_type { };
+NAMESPACE_END(std)
+
+
+NAMESPACE_BEGIN(drjit)
+struct dr_half {
+
+    uint16_t value;
+
+    #define DRJIT_IF_SCALAR template <typename Value, enable_if_t<std::is_arithmetic<Value>::value> = 0>
+
+    dr_half() = default;
+
+    DRJIT_IF_SCALAR dr_half(Value val) : value(float32_to_float16(float(val))) { }
```

Comment:
Using `float()` as a constructor rather than a C-style cast is a bit weird here. No?

Reply:
We should also maybe explicitly instantiate this constructor for all arithmetic types. Otherwise we won't get any implicit constructor calls like `dr_half my_var = 3u;` in other libraries.

Reply:
Fixed!